tasks/enable-repositories/CentOS.yml

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Enable CentOS repositories
+  command:
+    cmd: "dnf config-manager --set-enabled {{ item | quote }}"
+    # There doesn't seem to be a plugin for enabling already defined repos. So
+    # we silence the warning which recommends using an ansible plugin instead
+    # of running dnf command.
+    warn: no
+  loop: "{{ __ha_cluster_repos }}"
+  # The dnf command does not indicate if any changes have been made. To silence
+  # ansible lint (301 Commands should not change things if nothing needs doing),
+  # changed_when is set explicitly.
+  changed_when: yes
```

Comment:
> There doesn't seem to be a plugin for enabling already defined repos

Have you tried https://docs.ansible.com/ansible/2.8/modules/yum_repository_module.html#yum-repository-module ?

The use of `changed_when` breaks idempotency - the role will never be idempotent.  In order to be idempotent, if you cannot use `yum_repository` you'll have to use `command` or `shell` to check if the repos are enabled first, then only execute the `    cmd: "dnf config-manager --set-enabled {{ item | quote }}"` if the repo isn't enabled.

Reply:
Yes, I tried `yum repository`. It is not capable of changing an already existing repository.  For it to work, I had to write a complete repository definition. That approach seemed fragile to me as the HA repo file is different in various CentOS minor versions.

I added tasks to check if repositories are already enabled.